core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -149,6 +154,14 @@
   /** Whether to do URI scheme validation for file systems using this context.  */
   private boolean mUriValidationEnabled = true;
 
+  /** Cached map for workers. */
+  @GuardedBy("this")
+  private volatile List<BlockWorkerInfo> mWorkerInfoList = null;
```

Comment:
Should we just use the guava cache, like here: https://github.com/Alluxio/alluxio/pull/10963/files#diff-2d2ca0120802269babf6451df30b6279R273

Reply:
Agree. Let's at least keep a TODO. Leave it to @LuQQiu to decide if we want to switch to Guava Cache in this PR

Reply:
@gpang I am not quite familiar of guava cache, but according to my understanding, if I define `refreshAfterWrite(interval)`, it will refresh periodically no matter the value is being get or not. This makes sense on the leading master side since we want the cached worker info list up to date enough.

However, on the client-side, a client may need worker info list for some period of time and no need it anymore. Using loadingcache and refresh periodically may introduce unneeded RPC call to the leading master.